File: web/src/types/incident.ts
export type IncidentState = 'Detecting' | 'Active' | 'Resolved' | 'Archived'

export type AnomalyType =
  | 'CrashLoopBackOff'
  | 'OOMKilled'
  | 'ErrorExit'
  | 'RestartIncrement'
  | 'ImagePullBackOff'
  | 'CreateContainerConfigError'
  | 'FailedScheduling'
  | 'Evicted'
  | 'StateOscillation'

export interface Incident {
  id: string
  dedupKey: string
  state: IncidentState
  firstSeen: string
  lastSeen: string
  count: number
  namespace: string
  ownerKind: string
  ownerName: string
  anomalyType: AnomalyType
  message: string
  clusterId?: string
  podNames: string
  createdAt: string
  updatedAt: string
}

export interface Evidence {
  id: string
  incidentId: string
  type: string
  content: string
  error?: string
  collectedAt: string
  createdAt: string
}

export interface PaginatedResponse<T> {
  items: T[]
  total?: number
  page: number
  pageSize: number
  hasMore?: boolean
  nextCursorLastSeen?: string
  nextCursorId?: string
}

File: internal/domain/incident.go
package domain

import "time"

// IncidentState 事件状态
type IncidentState string

const (
	StateDetecting IncidentState = "Detecting"
	StateActive    IncidentState = "Active"
	StateResolved  IncidentState = "Resolved"
	StateArchived  IncidentState = "Archived"
)

// Incident 聚合后的事件实体（一次完整异常的生命周期）
type Incident struct {
	ID          string        `json:"id"`
	DedupKey    string        `json:"dedupKey"`
	State       IncidentState `json:"state"`
	FirstSeen   time.Time     `json:"firstSeen"`
	LastSeen    time.Time     `json:"lastSeen"`
	Count       int           `json:"count"`
	Namespace   string        `json:"namespace"`
	OwnerKind   string        `json:"ownerKind"`
	OwnerName   string        `json:"ownerName"`
	AnomalyType string        `json:"anomalyType"`
	Message     string        `json:"message"`
	PodNames    []string      `json:"podNames"`
	ClusterID   string        `json:"clusterId,omitempty"`
}
